### DISCUSSION TITLE
How to project `Schema` or `SchemaDescriptor` by `ProjectionMask`?

### DISCUSSION THREAD

**Discussion by RinChanNOWWW** [apache/arrow-rs#4695] [Q&A] [answered]
Suppose we have a parquet file whose schema is `(i Int, s Struct(a Int, b Struct(c int, d int)))`.

We can construct a `ProjectionMask { vec![true, false, false, true] }`  for `select i, s.b.d from t`, and the `Schema` of the output `RecordBatch` will be `(i Int, s Struct(b Struct(d int)))`.

My question is, how can I precompute the output `Schema` by the original `Schema` and the `ProjectionMask`?

I only found 

```rust
impl Schema {
    pub fn project(&self, indices: &[usize]) -> Result<Schema, ArrowError> {
      let new_fields = indices
          .iter()
          .map(|i| {
              self.fields.get(*i).cloned().ok_or_else(|| {
                  ArrowError::SchemaError(format!(
                      "project index {} out of bounds, max field {}",
                      i,
                      self.fields().len()
                  ))
              })
          })
          .collect::<Result<Vec<_>, _>>()?;
      Ok(Self::new_with_metadata(new_fields, self.metadata.clone()))
  }
}
```

which only can project root fields.

**Comment by tustvold** [comment, marked as answer]
I don't believe we support this directly on arrow Schema yet, but there is https://docs.rs/parquet/latest/parquet/arrow/fn.parquet_to_arrow_schema_by_columns.html

**Reply by RinChanNOWWW** [reply]
LGTM, thank you!!
